src/Cryptography/Sodium/Bindings/Scrypt.hs

```diff
@@ -0,0 +1,291 @@
+-- to lock memory regions storing plaintext passwords and to call
+-- 'Cryptography.Sodium.Bindings.SecureMemory.unlock' right after
+-- 'cryptoPWHashScryptSalsa2018SHA256Str' and 'cryptoPWHashScryptSalsa2018SHA256StrVerify'
+-- return.
+
+-- | Derive a key from a password.
+--
+-- For interactive, online operations, 'cryptoPWHashScryptSalsa2018SHA256OpsLimitInteractive'
+-- and 'cryptoPWHashScryptSalsa2018SHA256MemLimitInteractive' provide a safe baseline to be used.
+-- However, using higher values may improve security.
+--
+-- For highly sensitive data, 'cryptoPWHashScryptSalsa2018SHA256OpsLimitSensitive' and
+-- 'cryptoPWHashScryptSalsa2018SHA256MemLimitSensitive' can be used as an alternative.
+-- However, with these parameters, deriving a key takes about
+-- 2 seconds on a 2.8 GHz Core i7 CPU and requires up to 1 GiB of dedicated RAM.
+--
+-- The salt should be unpredictable. 'Cryptography.Sodium.Bindings.Random.randombytesBuf'
+-- is the easiest way to fill the 'cryptoPWHashScryptSalsa2018SHA256SaltBytes' bytes
+-- of the salt.
+--
+-- Keep in mind that to produce the same key from the same password, the same salt,
+-- opslimit, and memlimit values must be used. Therefore, these parameters must be stored for each user.
+--
+-- /See also:/ [crypto_pwhash_scryptsalsa208sha256()](https://doc.libsodium.org/advanced/scrypt#key-derivation)
+--
+-- @since 0.0.1.0
+foreign import capi "sodium.h crypto_pwhash_scryptsalsa208sha256"
+  cryptoPWHashScryptSalsa2018SHA256 ::
+    -- | A pointer to the computed key.
+    Ptr CUChar ->
+    -- | The length of the computer key.
+    -- Should be  between 'cryptoPWHashScryptSalsa2018SHA256BytesMin'
+    -- and 'cryptoPWHashScryptSalsa2018SHA256BytesMax' (~127 GB).
+    CULLong ->
+    -- | A pointer to the password from which the key is derived
+    Ptr CChar ->
+    -- | The length of the password.
+    -- Should be between 'cryptoPWHashScryptSalsa2018SHA256PasswdMin'
+    -- and 'cryptoPWHashScryptSalsa2018SHA256PasswdMax'.
+    CChar ->
+    -- | The salt, of length 'cryptoPWHashScryptSalsa2018SHA256SaltBytes'.
+    Ptr CUChar ->
+    -- | /opslimit:/ The maximum amount of computations to perform.
+    -- Must be between 'cryptoPWHashScryptSalsa2018SHA256OpsLimitMin'
+    -- and 'cryptoPWHashScryptSalsa2018SHA256OpsLimitMax'.
+    CULLong ->
+    -- | /memlimit:/ The maximum amount of RAM in bytes that the function will use.
+    -- It is highly recommended to allow the function to use at least 16 MiB.
+    -- This number must be between 'cryptoPWHashScryptSalsa2018SHA256MemLimitMin'
+    -- and 'cryptoPWHashScryptSalsa2018SHA256MemLimitMax'.
+    CSize ->
+    -- | Returns 0 on success, -1 if the computation didn't complete,
+    -- usually because the operating system refused to allocate the amount of
+    -- requested memory.
+    IO CInt
+
+----------------------
+-- Password Storage --
+----------------------
+
+-- | Generate an C ASCII encoded string which includes:
```

Comment:
'A' instead of 'an'. I also assume you mean that this generates a C string which is ASCII-encoded? I think 'an ASCII C-string' is clearer in that case.